BitcoinApp/CoinViewController/CoinNetworkService.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  CoinNetworkService.swift
+//  BitcoinApp
+//
+//  Created by Zhassulan Aimukhambetov on 12/4/19.
+//  Copyright © 2019 Zhassulan Aimukhambetov. All rights reserved.
+//
+
+import Foundation
+
+class CoinNetworkService {
+    
+    static private let urlApi: String = "https://api.coindesk.com/v1/bpi/currentprice/"
+    
+    static func getCoin (currency: Currency, completion: @escaping (Coin) -> Void) {
+        let urlString = urlApi + currency.name + ".json"
+        guard let url = URL(string: urlString) else { return }
+        NetworkService.shared.getData(url: url) { (json) in
+            guard let json = json as? [String : AnyObject] else { return }
+            guard let base = json["bpi"] as? [String : AnyObject] else { return }
```

Comment:
Your massive early returns is a common serious mistake. In this failure case, your completion would never be called. Imagine a Loading Indicator which take a whole screen and waits for your callback to get rid of screen. It would be stuck if don't get a callback at all.

Reply:
All your early returns should call Callback with some Error. Read the Apple Swift book to learn more about errors

Reply:
Also, it's a common case to use `completion: (Result<Coin, Error>) -> Void`